network/crl.go

```diff
@@ -331,23 +334,36 @@ func (v DefaultCRLVerifier) getCachedOrFetch(url string, allowLocal bool, issuer
 		return nil, err
 	}
 
+	revokedCertificates := make(map[*big.Int]pkix.RevokedCertificate, len(crl.TBSCertList.RevokedCertificates))
+	for _, cert := range crl.TBSCertList.RevokedCertificates {
+		revokedCertificates[cert.SerialNumber] = cert
+	}
+
 	if crl.TBSCertList.NextUpdate.Before(time.Now()) {
 		log.Warnf("CRL: CRL at %s is outdated", url)
 		return nil, ErrOutdatedCRL
 	}
 
+	// We don't need this array anymore as all revoked certificates are now inside the `map`
+	crl.TBSCertList.RevokedCertificates = make([]pkix.RevokedCertificate, 0)
```

Comment:
Do we really need to initialize a new empty slice. I think you can just set it to nil or you are going to reuse it somewhere else?

Reply:
Tried `= []`, forgot about `= nil`, thanks, will fix